packages/rum-react/test/appendComponent.ts

```diff
@@ -0,0 +1,22 @@
+import { createRoot } from 'react-dom/client'
+import { flushSync } from 'react-dom'
+import { noop } from '@datadog/browser-core'
+import { appendElement } from '../../rum-core/test'
+import { registerCleanupTask } from '../../core/test'
+
+export function appendComponent(component: React.ReactNode) {
+  const container = appendElement('<div></div>')
+  const root = createRoot(container, {
+    // Do nothing by default when an error occurs
+    onRecoverableError: noop,
+  })
+  flushSync(() => {
+    root.render(component)
+  })
+  registerCleanupTask(() => {
+    flushSync(() => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why is this flushSync required in the cleanupTask?

Reply:
Right, I was thinking `root.unmount()` was asynchronous, but after more tests it seems like it isn't so `flushSync` is probably not needed. I'll remove it.